storage/maria/ma_backup.c

```diff
@@ -199,14 +199,17 @@ int aria_read_index(File kfile, ARIA_TABLE_CAPABILITIES *cap, ulonglong block,
       length= _ma_get_page_used(&share, buffer);
       if (length > cap->block_size - CRC_SIZE)
         DBUG_RETURN(HA_ERR_CRASHED);
-      error= maria_page_crc_check(buffer, block, &share,
-                                  MARIA_NO_CRC_NORMAL_PAGE,
-                                  (int) length);
-      if (error != HA_ERR_WRONG_CRC)
+      if (!_ma_check_if_zero(buffer, share.block_size - CRC_SIZE))
+        error= 0;
+      else
+        error= maria_page_crc_check(buffer, block, &share,
+                                    MARIA_NO_CRC_NORMAL_PAGE,
+                                    (int) length);
+      if (error == 0 || my_errno != HA_ERR_WRONG_CRC)
         DBUG_RETURN(error);
```

Comment:
882f6fa3aa320ffcca19c883b652d12e7b31812b is not adding any call to `_ma_check_if_zero` here; it is adding that to `aria_read_data()` only.

Reply:
I didn't face the issue in `aria_read_index()` . I added just to be over protective.